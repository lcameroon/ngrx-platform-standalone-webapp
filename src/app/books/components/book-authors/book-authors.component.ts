import { Component, Input } from '@angular/core';

import { Book } from '../../models/book';

@Component({
    selector: 'app-book-authors',
    templateUrl: './book-authors.component.html',
    styleUrls: ['book-authors.component.scss']
})
export class BookAuthorsComponent {
    @Input() book: Book;

    get authors() {
        return this.book.volumeInfo.authors;
    }
}
