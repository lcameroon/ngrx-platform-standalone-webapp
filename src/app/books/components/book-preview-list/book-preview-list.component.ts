import { Component, Input } from '@angular/core';
import { Book } from '../../models/book';

@Component({
    selector: 'app-book-preview-list',
    templateUrl: './book-preview-list.component.html',
    styleUrls: ['book-preview-list.component.scss']
})
export class BookPreviewListComponent {
    @Input() books: Book[];
}
