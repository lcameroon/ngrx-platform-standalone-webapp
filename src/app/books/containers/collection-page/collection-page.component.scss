// 
// Container components are permitted to have just enough styles
// to bring the view together. If the number of styles grow,
// consider breaking them out into presentational
// components.
// 
mat-card-title {
    display: flex;
    justify-content: center;
}