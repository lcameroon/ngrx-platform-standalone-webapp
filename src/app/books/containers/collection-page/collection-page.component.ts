import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
import { select, Store } from '@ngrx/store';
import { Observable } from 'rxjs';

import { Book } from '../../models/book';
import * as CollectionActions from '../../actions/collection.actions';
import * as fromBooks from '../../reducers';

@Component({
    selector: 'app-collection-page',
    changeDetection: ChangeDetectionStrategy.OnPush,
    templateUrl: './collection-page.component.html',
    styleUrls: ['collection-page.component.scss']
})
export class CollectionPageComponent implements OnInit {
    books$: Observable<Book[]>;

    constructor(private store: Store<fromBooks.State>) {
        this.books$ = store.pipe(select(fromBooks.getBookCollection));
    }

    ngOnInit() {
        this.store.dispatch(new CollectionActions.Load());
    }
}
