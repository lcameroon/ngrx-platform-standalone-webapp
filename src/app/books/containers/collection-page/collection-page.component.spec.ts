import { combineReducers, Store, StoreModule } from '@ngrx/store';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { MatCardModule, MatInputModule } from '@angular/material';

import { CollectionPageComponent } from './collection-page.component';
import { BookPreviewListComponent } from '../../components/book-preview-list/book-preview-list.component';
import { BookPreviewComponent } from '../../components/book-preview/book-preview.component';
import { BookAuthorsComponent } from '../../components/book-authors/book-authors.component';
import { EllipsisPipe } from '../../../shared/pipes/ellipsis.pipe';
import { AddCommasPipe } from '../../../shared/pipes/add-commas.pipe';

import * as CollectionActions from '../../actions/collection.actions';
import * as fromBooks from '../../reducers';

describe('Collection Page', () => {
    let fixture: ComponentFixture<CollectionPageComponent>;
    let store: Store<fromBooks.State>;
    let instance: CollectionPageComponent;

    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [
                NoopAnimationsModule,
                StoreModule.forRoot({
                    books: combineReducers(fromBooks.reducers)
                }),
                MatCardModule,
                MatInputModule,
                RouterTestingModule
            ],
            declarations: [
                CollectionPageComponent,
                BookPreviewListComponent,
                BookPreviewComponent,
                BookAuthorsComponent,
                AddCommasPipe,
                EllipsisPipe
            ]
        });

        fixture = TestBed.createComponent(CollectionPageComponent);
        instance = fixture.componentInstance;
        store = TestBed.get(Store);

        spyOn(store, 'dispatch').and.callThrough();
    });

    it('should compile', () => {
        fixture.detectChanges();

        expect(fixture).toMatchSnapshot();
    });

    it('should dispatch a collection.Load on init', () => {
        const action = new CollectionActions.Load();

        fixture.detectChanges();

        expect(store.dispatch).toHaveBeenCalledWith(action);
    });
});
