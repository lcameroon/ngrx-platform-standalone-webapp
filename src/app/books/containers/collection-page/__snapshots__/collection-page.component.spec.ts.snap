// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection Page should compile 1`] = `undefined`;
